functions/lib/stripe/checkout.py

```diff
@@ -0,0 +1,157 @@
+    logger.error(f"Provided event {event_ref.path} does not exist in the database. Returning status=404")
+    return json.dumps({"url": ERROR_URL})
+  
+  event = maybe_event.to_dict()
+  
+  # 1. check for event is stripe enabled
+  if (not event.get("paymentsActive")):
+    print("Payments not active")
+    logger.error(f"Provided event {event_ref.path} does not have payments enabled. Returning status=500")
+    return json.dumps({"url": cancel_url})
+
+  organiser_id = event.get("organiserId")
+  logger.info(f"Event with id {event_id} has organiser with id {organiser_id}.")
+  organiser_ref = db.collection("Users").document(organiser_id)
+  maybe_organiser = organiser_ref.get(transaction=transaction)
+  
+  # Check if the organiser exists, if not error out
+  if (not maybe_organiser.exists):
+    logger.error(f"Provided event {event_ref.path} has an organiser {organiser_ref.path} who does not exist in the database. Returning status=404")
+    return json.dumps({"url": ERROR_URL})
+
+  organiser = maybe_organiser.to_dict()
+
+  # 2a. check for event organiser has stripe account
+  if (organiser.get("stripeAccount") == None):
+    logger.error(f"Provided event {event_ref.path} has an organiser {organiser_ref.path} who does not have a stripe account. Returning status=500")
+    return json.dumps({"url": ERROR_URL})
+  
+  # 2b. check if the stripe account is active
+  if (organiser.get("stripeAccountActive") == False):
+    account = stripe.Account.retrieve(organiser.get("stripeAccount"))
+    # if its not active, double check to see if this is the n+1 iteration, hence if they have charges enabled and details submitted for stripe, open their account, else error out
+    if account.charges_enabled and account.details_submitted:
+      transaction.update(organiser_ref, {"stripeAccountActive": True})
+      logger.info(f"Provided organiser {organiser_ref.path} already has all charges enabled and details submitted. Activiating their sportshub stripe account.")
+
+    else:
+      logger.error(f"Provided event {event_ref.path} has an organiser {organiser_ref.path} who does not have a active stripe account. charges_enabled={account.charges_enabled} details_submitted={account.details_submitted} Returning status=500")
+      return json.dumps({"url": ERROR_URL})
+
+  # 3. check again with database in transction in the backend if quantity is still available
+  vacancy = event.get("vacancy")
+
+  if vacancy < quantity:
+    logger.warning(f"Provided event {event_ref.path} does not have enough tickets to fulfill this order. quantity_requested={quantity} vacancy={vacancy}")
+    return json.dumps({"url": cancel_url})
+  
+  # 4. obtain price and organiser stripe account id again from db
+  price = event.get("price")
+  organiser_stripe_account_id = organiser.get("stripeAccount")
+
+  # 4a. check if the price exists for this event
+  if (price == None or not isinstance(price, int) or price <= 1): # we don't want events to be less than stripe fees
```

Comment:
should be a float instead of an interger as price cna be 12.50

Reply:
this is a very good catch, thank you!! let me change it for the interim. However we should move away from using floating point numbers all together... they are hard to operate on and introduce rounding errors.

we should start storing the price as the most granular measurement our currency offers, which is cents in $AUD. That way we can process everything as a integer, and then just convert it to dollars just for the UI. Furthermore, stripe also operates in cents.

E.g. a $15.50 ticket will be stored as 1550... and we just default all our calculations to manipulation of cents.

let me make a ticket for it.